imap_processing/glows/l1/glows_l1a.py

```diff
@@ -0,0 +1,405 @@
+    for hist in hist_l0:
+        hist_l1a = HistogramL1A(hist)
+        # Split by IMAP start time
+        # TODO: Should this be MET?
+        hist_day = calc_start_time(hist.SEC).astype("datetime64[D]")
+        hists_by_day[hist_day].append(hist_l1a)
+
+    # Generate CDF files for each day
+    generated_files = []
+    for hist_l1a_list in hists_by_day.values():
+        dataset = generate_histogram_dataset(hist_l1a_list, data_version)
+        generated_files.append(write_cdf(dataset))
+
+    for de_l1a_list in de_by_day.values():
+        dataset = generate_de_dataset(de_l1a_list, data_version)
+        generated_files.append(write_cdf(dataset))
+
+    return generated_files
+
+
+def process_de_l0(
+    de_l0: list[DirectEventL0],
+) -> dict[np.datetime64, list[DirectEventL1A]]:
+    """
+    Process Direct Event packets into GLOWS L1A CDF files.
+
+    This involves combining packets with direct event sequences that span multiple
+    packets.
+
+    Parameters
+    ----------
+    de_l0 : list[DirectEventL0]
+        List of DirectEventL0 objects
+
+    Returns
+    -------
+    de_by_day : dict[np.datetime64, list[DirectEventL1A]]
+        Dictionary with keys of days and values of lists of DirectEventL1A objects.
+        Each day has one CDF file associated with it.
+    """
+    de_by_day = dict()
+
+    for de in de_l0:
+        de_day = calc_start_time(de.MET).astype("datetime64[D]")
+        if de_day not in de_by_day:
+            de_by_day[de_day] = [DirectEventL1A(de)]
+        elif de.SEQ != 0:
+            # If the direct event is part of a sequence and is not the first,
+            # append it to the last direct event in the list
+            de_by_day[de_day][-1].append(de)
+        else:
+            de_by_day[de_day].append(DirectEventL1A(de))
+
+    return de_by_day
+
+
+def generate_de_dataset(
+    de_l1a_list: list[DirectEventL1A], data_version: str
+) -> xr.Dataset:
+    """
+    Generate a dataset for GLOWS L1A direct event data CDF files.
+
+    Parameters
+    ----------
+    de_l1a_list : list[DirectEventL1A]
+        List of DirectEventL1A objects for a given day
+    data_version : str
+        Data version for CDF filename, in the format "vXXX"
+
+    Returns
+    -------
+    output : xr.Dataset
+        Dataset containing the GLOWS L1A direct event CDF output
+    """
+    # TODO: Block header per second, or global attribute?
+
+    time_data = np.zeros(len(de_l1a_list), dtype="datetime64[ns]")
+    # TODO: Should each timestamp point to a list of direct events, each with a
+    #  timestamp? Or should the list be split out to make the timestamps?
+
+    # Each DirectEventL1A class covers 1 second of direct events data
+    direct_events = np.zeros((len(de_l1a_list), len(de_l1a_list[0].direct_events), 4))
+    global_attributes = glows_cdf_attrs.glows_l1a_de_attrs.output()
+
+    global_attributes["ground_software_version"] = __version__
```

Comment:
My first question was related to this. I was wondering if you could do similar to this line and avoid having to pass data_version parameter.